src/audio/copier/copier.c

```diff
@@ -1007,7 +1219,24 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 	if (cd->endpoint_num) {
 		for (i = 0; i < cd->endpoint_num; i++) {
 			update_internal_comp(dev, cd->endpoint[i]);
-			ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i], params);
+
+			/* For ALH multi-gateway case, params->channels is a total multiplexed
+			 * number of channels. Demultiplexed number of channels for each individual
+			 * gateway comes in blob's struct ipc4_alh_multi_gtw_cfg.
```

Comment:
in 100% of the cases today, we end-up sending the SAME information across multiple links.

All these channel changes are really useless in practice, at least for playback.

Reply:
Not sure if I understand the comment correctly. This change here is required: when multiplexing/demultiplexing of multiple links is used, struct sof_ipc_stream_params contains multiplexed stream channel count, individual links will have less channels. Channel count for such demultiplexed links is received from channel map found in struct ipc4_alh_multi_gtw_cfg (comes in blob).
If ALH multi-gateway feature is not used, we only execute one "if" in copier_copy(), so the changes does not really affect performance in this case.